website/docs/components/data-connectors/mongodb.md

```diff
@@ -0,0 +1,240 @@
+```json
+{
+  "a": 1,
+  "b": {
+    "x": 2,
+    "y": {
+      "z": 3
+    }
+  }
+}
+```
+
+Using `mongodb_unnest_depth` you can control the unnesting behavior. Here are the examples:
+
+### mongodb_unnest_depth: 0
+```sql
+sql> select * from test_table;
++-----------+---------------------+
+| a (Int32) | b (Utf8)            |
++-----------+---------------------+
+| 1         | {"x":2,"y":{"z":3}} |
++---+-----------------------------+
+```
+
+### mongodb_unnest_depth: 1
+```sql
+sql> select * from test_table;
++-----------+-------------+------------+
+| a (Int32) | b.x (Int32) | b.y (Utf8) |
++-----------+-------------+------------+
+| 1         | 2           | {"z":3}    | 
++-----------+-------------+------------+
+```
+
+### mongodb_unnest_depth: 2
+```sql
+sql> select * from test_table;
++-----------+-------------+---------------+
+| a (Int32) | b.x (Int32) | b.y.z (Int32) |
++-----------+-------------+---------------+
+| 1         | 2           | 3             |
++-----------+-------------+---------------+
+```
+
+## Examples
+
+### Connecting using username and password and custom auth table
+
+```yaml
+datasets:
+  - from: mongodb:my_dataset
+    name: my_dataset
+    params:
+      mongodb_host: localhost
+      mongodb_port: 27017
+      mongodb_db: my_database
+      mongodb_user: my_user
+      mongodb_pass: ${secrets:mongodb_pass}
+      mongodb_auth_source: admin
+```
+
+### Connecting using SSL
+
+```yaml
+datasets:
+  - from: mongodb:my_dataset
+    name: my_dataset
+    params:
+      mongodb_host: localhost
+      mongodb_port: 27017
+      mongodb_db: my_database
+      mongodb_user: my_user
+      mongodb_pass: ${secrets:mongodb_pass}
+      mongodb_sslmode: preferred
+      mongodb_sslrootcert: ./custom_cert.pem
+```
+
+### Connecting using a Connection String
+
+```yaml
+datasets:
+  - from: mongodb:my_dataset
+    name: my_dataset
+    params:
+      mongodb_connection_string: mongodb://${secrets:my_user}:${secrets:my_password}@localhost:27017/my_db?authSource=admin
+```
+
+
+### With custom connection pool settings
+
+```yaml
+datasets:
+  - from: mongodb:my_dataset
+    name: my_dataset
+    params:
+      mongodb_host: localhost
+      mongodb_port: 27017
+      mongodb_db: my_database
+      mongodb_user: my_user
+      mongodb_pass: ${secrets:mongodb_pass}
+      mongodb_pool_min: 5
+      mongodb_pool_max: 10
+```
+
+## Secrets
+
+Spice integrates with multiple secret stores to help manage sensitive data securely. For detailed information on supported secret stores, refer to the [secret stores documentation](/docs/components/secret-stores). Additionally, learn how to use referenced secrets in component parameters by visiting the [using referenced secrets guide](/docs/components/secret-stores#using-secrets).
+
+## Cookbook
+
+- A cookbook recipe to configure MongoDB as a data connector in Spice. [MongoDB Data Connector](https://github.com/spiceai/cookbook/tree/trunk/mongodb/connector#readme)
```

Comment:
Cookbook will need to be added to the cookbook index for https://spiceai.org/cookbook

Reply:
> Or is there anything else that needs to be done?
@krinart recipe needs to be added here https://github.com/spiceai/docs/blob/trunk/website/src/pages/cookbook/_ignored.tsx

Reply:
Added in this PR https://github.com/spiceai/docs/pull/1116